src/emit.ts

```diff
@@ -10,43 +11,44 @@ let events: Record<string, typeof componentEvents>
 export function emitted<T = unknown>(
   vm: ComponentPublicInstance,
   eventName?: string
-): T[] | Record<string, T[]> {
+): undefined | T[] | Record<string, T[]> {
   const cid = vm.$.uid
 
   const vmEvents = (events as Record<string, Record<string, T[]>>)[cid] || {}
   if (eventName) {
-    const emitted = vmEvents
-      ? (vmEvents as Record<string, T[]>)[eventName]
-      : undefined
-    return emitted
+    return vmEvents ? (vmEvents as Record<string, T[]>)[eventName] : undefined
   }
 
   return vmEvents as Record<string, T[]>
 }
+type Events = { [id: number]: Record<string, any> }
```

Comment:
Not sure on the best practice here but this could be `<Record<number, Record<string, any>>`?

Also, perhaps this should actually be `unknown`?